core/src/protocols/stream/quic/mod.rs

```diff
@@ -39,6 +40,9 @@ pub struct QuicPacket {
 
     /// The number of bytes contained in the estimated payload
     pub payload_bytes_count: Option<u64>,
+
+    // The decrypted QUIC packet payload
+    pub decrypted_payload: Option<Vec<u8>>,
```

Comment:
We had a few questions out of curiosity - not feedback - 

I ran this, and got a dump of raw payload (just bytes). Can you clarify what is expected to be in that payload? Is there anything that should be parsed from it, for example that a researcher might reasonably want to filter on? If not, what kinds of possible use-cases are there for collecting the raw bytes?

Reply:
Yes, raw bytes are definitely not helpful! 

For background, I have an ongoing project looking at specific features of the QUIC Client Init that can be used to develop a fingerprint and identify QUIC clients. I already have a functioning QUIC parser that was built on top of [TLS Fingerprint](https://github.com/refraction-networking/tls-fingerprint). However, I have been wanting to rebuild the QUIC version on top of Retina. 

Before this is ready to merge I intend to add support for a fully parsed QUIC Client Init including the TLS ClientHello.  I added that field temporarily to ensure that the decrypted payload was as expected, a list of QUIC frames.

To answer your question, I envision a researcher being able to filter on the same things one could filter in TLS like SNI.

Reply:
The latest commit introduces frame parsing. I have replaced the `decrypted_payload` with a `frames` vector on the `QuicPacket` struct. Once I have the CRYPTO frame reassembly and TLS ClientHello parsing implemented I will remove the data field from the serialized CRYPTO frame output which will help clean things up further.